File: laundry-reservation/typescript/tests/FixedPinGenerator.ts
import type { PinGenerator } from '../src/PinGenerator.js';

export class FixedPinGenerator implements PinGenerator {
  private readonly pins: number[];
  private index = 0;

  constructor(...pins: number[]) {
    this.pins = pins;
  }

  generate(): number {
    return this.pins[this.index++]!;
  }
}

File: laundry-reservation/csharp/tests/LaundryReservation.Tests/FixedPinGenerator.cs
namespace LaundryReservation.Tests;

public class FixedPinGenerator : IPinGenerator
{
    private readonly Queue<int> _pins;

    public FixedPinGenerator(params int[] pins)
    {
        _pins = new Queue<int>(pins);
    }

    public int Generate() => _pins.Dequeue();
}
